### PR TITLE
Add logo script phishing url script

### PR DESCRIPTION
**Pull request by yguezpa**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/45495) [link to the issue](https://github.com/demisto/etc/issues/47445 ) [link to the issue](https://github.com/demisto/etc/issues/47446)

### CODE REVIEW

**Comment by ShirleyDenkberg** [comment]
@erezh31 @Itay4 @Noy-Maimon Doc review completed.
@yguezpa Need a README for the new script.

File: Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+            minor, model = get_minor_version_upgrade(model)
+            msg = save_upgraded_version_model(model)
+            msg_list.append(msg)
+        else:
+            return_error(msg)
+
+    # Case where there were new new model release -> load model from demisto
+    elif (demisto_major_version == MAJOR_VERSION):
+        model = load_demisto_model()
+        if action == KEY_ADD_LOGO:
```
Comment (erezh31) [inline Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py:244]: code duplication - seems like line 227-239 and 244-256 are totally the same
you can use the if just to load the model and then preform the extact same logic for adding \ removing logo or just extract this logic to a function

File: Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+    # Case where new model release and logo were added -> transfer logo from model in demisto to new model in docker
+    elif (demisto_major_version < MAJOR_VERSION) and (demisto_minor_version > MINOR_DEFAULT_VERSION):
+        model_docker = load_model_from_docker()
+        model_demisto = load_demisto_model()
+        model_docker.logos_dict = model_demisto.logos_dict
+        model_docker.custom_logo_associated_domain = model_demisto.custom_logo_associated_domain
+        model_docker.top_domains = model_demisto.top_domains
+        msg_list.append(MSG_TRANSFER_LOGO)
+        if action == KEY_ADD_LOGO:
```
Comment (erezh31) [inline Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py:266]: again seems like same logic - line 266-280 - use function you will create

**Review by erezh31 — CHANGES_REQUESTED** [review]

File: Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py [hunk 3]
```diff
@@ -503,44 +509,58 @@ def extract_created_date(entry_list: List):
...
+        return create_dict_context(url, url, MSG_INVALID_URL % (error, status_code), {}, SCORE_INVALID_URL,
+                                   is_white_listed,
                                    {})
 
+    # Rasterize html and image
+    res_rasterize = demisto.executeCommand('rasterize', {'type': 'json',
+                                                         'url': url,
+                                                         'wait_time': WAIT_TIME_RASTERIZE
+                                                         })
+    if KEY_IMAGE_RASTERIZE not in res_rasterize[0]['Contents'].keys() or KEY_IMAGE_HTML not in res_rasterize[0][
```
[the remark this answers is not in the document]
Comment (yguezpa) [reply Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py:526]: is_valid_url is a function that checks if url is valid and is runned before rasterize

File: Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  default: false
+  description: Action to execute on the model. Will be ignored if displayLogos
+    is set to True.
+  isArray: false
+  name: action
+  predefined:
+  - RemoveLogo
+  - AddLogo
+  - ModifiedDomainForLogo
+  required: false
```
[the remark this answers is not in the document]
Comment (yguezpa) [reply Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.yml:54]: I have a check that return a message if no action but can add mandatory as well

[the remark this answers is not in the document]
Comment (erezh31) [reply Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.yml:54]: please add it as well, it's important and will prevent errors with the UI - it makes sure the user fill it before execution

File: Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (erezh31) [reply Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py:526]: ok, sounds fair

File: Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py [hunk 5]
```diff
@@ -503,44 +509,58 @@ def extract_created_date(entry_list: List):
...
+    if len(res_rasterize) > 0:
+        output_rasterize = res_rasterize[0]['Contents']
+    else:
+        raise DemistoException(MSG_FAILED_RASTERIZE)
+    if isinstance(output_rasterize, str):
+        raise DemistoException(MSG_FAILED_RASTERIZE)
+
+    # Get final url and redirection
+    final_url = output_rasterize.get('current_url')
+    if final_url != url:
```
[the remark this answers is not in the document]
Comment (yguezpa) [reply Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py:538]: What do you mean if the rasterize did not update. I handle if url not working. If url working there should always be a url in the rasterize no ?

File: Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py (hunk 2 above)
Comment (yguezpa) [reply Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py:266]: not exactly the same but I created 1 function

File: Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py (hunk 1 above)
Comment (yguezpa) [reply Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py:244]: not exactly the same but I created 1 function

File: Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+    if (action == KEY_ADD_LOGO) and (not logo_image_id or not logo_name):
+        return_error(MSG_EMPTY_NAME_OR_URL)
+    elif not logo_name:
+        return_error(MSG_EMPTY_LOGO_NAME)
+
+    if action==KEY_ADD_LOGO:
+        res = demisto.getFilePath(logo_image_id)
+        path = res['path']
+        with open(path, 'rb') as file:
+            logo_content = file.read()
```
[the remark this answers is not in the document]
Comment (yguezpa) [reply Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py:220]: Yes we already do that. It's inside the model pickle so you don't see it
Screen Shot 2022-03-23 at 15 13 23
.

File: Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (erezh31) [reply Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py:538]: rasterize integration

**Review by DeanArbel — APPROVED** [review]
Nice.

**Pull request merged**